### ISSUE TITLE
terraform-docs produces output that fails terrafmt fmt check

### ISSUE THREAD

**Issue by matt-FFFFFF** [Azure/tfmod-scaffold#9]
Meeting with @grayzu resulted in this issue

I use `terraform-docs` and `terrafmt` on this module: https://github.com/Azure/terraform-azurerm-lz-vending

I had an issue where the README.md produced by terraform-docs failed the terrafmt fmt check.

The issue was when terraform-docs outputs the type declaration for some complex inputs.

As an example, see here: https://github.com/Azure/terraform-azurerm-lz-vending/blob/main/README.md?plain=1#L221-L231

terraform-docs outputs the following Markdown for the type declaration of the input variable `role_assignments`:

````markdown
Type:

```hcl
list(object({
    principal_id   = string,
    definition     = string,
    relative_scope = string,
  }))
```

Default: `[]`
````

This fails the terrafmt fmt test!

```
ERRO[2023-01-12 12:02:01] block 4 @ README.md:223 failed to process with: failed to parse hcl: README.md:1,1-5: Argument or block definition required; An argument or block definition is required here. To set an argument, use the equals sign "=" to introduce the argument value.
list(object({
    principal_id   = string,
    definition     = string,
    relative_scope = string,
  }))
```

I had to implement a workaround to avoid terrafmt checking the generated README.md file.
I placed this into the Makefile (note the `grep -v README.md`):

```
fmt:
        @echo "==> Fixing source code with gofmt..."
        find ./tests -name '*.go' | grep -v vendor | xargs gofmt -s -w
        @echo "==> Fixing Terraform code with terraform fmt..."
        terraform fmt -recursive
        @echo "==> Fixing embedded Terraform with terrafmt..."
        find . | egrep ".md|.tf" | grep -v README.md | sort | while read f; do terrafmt fmt $$f; done
```

I would suggest that we need to do similar in the tfmod container to avoid these issues.

/cc @jtracey93

**Comment by lonegunmanb** [comment]
Thanks @matt-FFFFFF for opening this issue, I've updated scaffold to fix this issue at #10, would you please try again to confirm? Thanks!

**Comment by matt-FFFFFF** [comment]
Looks great - thanks I will test it shortly
